File: accsaber-api/src/main/java/de/ixsen/accsaber/api/mapping/MappingComponent.java
package de.ixsen.accsaber.api.mapping;

import org.springframework.stereotype.Component;

@Component
public class MappingComponent {

    public PlayerMapper getPlayerMapper() {
        return PlayerMapper.INSTANCE;
    }

    public RankedMapMapper getRankedMapMapper() {
        return RankedMapMapper.INSTANCE;
    }

    public MapLeaderboardPlayerMapper getMapLeaderboardPlayerMapper() {
        return MapLeaderboardPlayerMapper.INSTANCE;
    }

    public ScoreMapper getScoreMapper() {
        return ScoreMapper.INSTANCE;
    }
}

File: accsaber-api/src/main/kotlin/de/ixsen/accsaber/api/mapping/MappingComponent.kt
package de.ixsen.accsaber.api.mapping

import org.springframework.stereotype.Component

@Component
class MappingComponent {
    val playerMapper: PlayerMapper
        get() = PlayerMapper.Companion.INSTANCE
    val rankedMapMapper: RankedMapMapper
        get() = RankedMapMapper.Companion.INSTANCE
    val mapLeaderboardPlayerMapper: MapLeaderboardPlayerMapper
        get() = MapLeaderboardPlayerMapper.Companion.INSTANCE
    val scoreMapper: ScoreMapper
        get() = ScoreMapper.Companion.INSTANCE
}